File: Lista-15/go/14.go
package main
import "sync"
import "fmt"

var (
    instance *Configuracao
    once     sync.Once
)

type Configuracao struct{}

func GetInstancia() *Configuracao {
    once.Do(func() {
        instance = &Configuracao{}
    })
    return instance
}

func main() {
    config1 := GetInstancia()
    config2 := GetInstancia()
    fmt.Println(config1 == config2)  // Saída: true (mesma instância)
}

File: Lista-15/python/14.py
class Configuracao:
    _instancia = None

    def __new__(cls):
        if cls._instancia is None:
            cls._instancia = super(Configuracao, cls).__new__(cls)
        return cls._instancia

# Exemplo de uso
config1 = Configuracao()
config2 = Configuracao()
print(config1 is config2)  # Saída: True (mesma instância)
